packages/ember-runtime/lib/system/object.js

```diff
@@ -21,7 +24,48 @@ import { DEBUG } from 'ember-env-flags';
   @uses Ember.Observable
   @public
 */
-const EmberObject = CoreObject.extend(Observable);
+const EmberObject = CoreObject.extend(Observable, {
+  _debugContainerKey: descriptor({
+    enumerable: false,
+    get() {
+      if (this[OVERRIDE_CONTAINER_KEY]) {
+        return this[OVERRIDE_CONTAINER_KEY];
+      }
+
+      let meta = peekMeta(this);
+      let factory  = meta.getFactory();
+
+      return factory && factory.fullName;
+    },
+
+    // we need a setter here largely to support the legacy
+    // `owner._lookupFactory` and its double extend
+    set(value) {
+      this[OVERRIDE_CONTAINER_KEY] = value;
+    }
+  }),
+
+  [OWNER]: descriptor({
+    enumerable: false,
+    get() {
+      if (this[OVERRIDE_OWNER]) {
+        return this[OVERRIDE_OWNER];
+      }
+
+      let meta = peekMeta(this);
+      let factory  = meta.getFactory();
+
+      return factory && factory.owner;
+    },
+
+    // we need a setter here largely to support the legacy
```

Comment:
ah, so this `[OVERRIDE_OWNER]` both in the setter and getter can be removed once the legacy support is dropped?